proto/cacheclient.proto

```diff
@@ -48,3 +51,39 @@ message _SetResponse {
   ECacheResult result = 1;
   string message = 2;
 }
+
+message _HashGetRequest {
+  bytes cache_hash_name = 1;
+  bytes cache_hash_key = 2;
+}
+
+message _HashGetResponse {
+  ECacheResult result = 1;
+  bytes cache_body = 2;
+  string message = 3;
+}
+
+message _HashGetAllRequest {
+  bytes cache_hash_name = 1;
+}
+
+message HashKeyValuePair {
+  bytes key = 1;
+  bytes value = 2;
+}
+
+message _HashGetAllResponse {
+  ECacheResult result = 1;
+  repeated HashKeyValuePair cache_body = 2;
+  string message = 3;
+}
+
+message _HashSetRequest {
+  bytes cache_hash_name = 1;
+  repeated HashKeyValuePair mapping = 2;
```

Comment:
With this type you'll also need to provide a hash_delete() api with a key list

Reply:
Depending how much granularity we want:
(1) delete the entire hash from momento (no key list required)
(2) delete a list of keys

Reply:
Deleting the entire hash will prooooobably be done via the scs delete api. I think you're looking at needing (2).

Reply:
I'll leave this for a separate PR. Current ask if for Get, Set, and GetAll.